src/app/(home)/globe/index.tsx

```diff
@@ -55,6 +77,26 @@ function GlobeInner() {
     }
   }, []);
 
+  const onMouseMove = useCallback(
+    (event: MapLayerMouseEvent) => {
+      if (!mapRef.current) return;
+      const features = mapRef.current.queryRenderedFeatures(event.point, {
+        layers: ["area-clickable-polygon"],
+      });
+
+      const interactiveItem = features && features[0];
+      if (interactiveItem) {
+        setHighlightArea({
+          ...interactiveItem.properties,
+          popupLocation: event.lngLat,
+        } as IHighligthArea);
+      } else {
+        setHighlightArea(undefined);
+      }
+    },
+    [setHighlightArea]
+  );
```

Comment:
@oliverroick I suggest to throttle or debounce this function call.

Reply:
I addressed this by putting the GeoJSON Feature for the highlighted polygon in state, and then I added new source and layer for the highlight. That way we don't have to run the filter across the area-boundary data every time the mouse moves.

Reply:
Great! This is much better now.